File: 2022/src/test/kotlin/Day18Test.kt
import org.junit.Assert.*
import org.junit.Test
import utils.getResourceAsString

class Day18Test {

  @Test
  fun part1_sample() {
    assertEquals(64, Day18.part1(getResourceAsString("day18-sample.txt")))
  }

  @Test
  fun part1() {
    assertEquals(4628, Day18.part1(getResourceAsString("day18.txt")))
  }

  @Test
  fun part2_sample() {
    assertEquals(58, Day18.part2(getResourceAsString("day18-sample.txt")))
  }

  @Test
  fun part2() {
    assertEquals(2582, Day18.part2(getResourceAsString("day18.txt")))
  }

}

File: 2018/test/day18.test.ts
import fs from "fs";
import assert from "assert";
import { part1, part2 } from "../src/day18";

describe("Day 18", function () {
  it("Part 1 - Sample", function () {
    const input = readInput("input/day18-sample.txt");
    assert.equal(part1(input), 1147);
  });

  it("Part 1", function () {
    const input = readInput("input/day18.txt");
    assert.equal(part1(input), 514944);
  });

  it("Part 2", function () {
    const input = readInput("input/day18.txt");
    assert.equal(part2(input), 193050);
  });
});

function readInput(path: string) {
  return fs.readFileSync(path).toString();
}
